### ISSUE TITLE
Stdlib: Add Map.filter & Map.reject

### ISSUE THREAD

**Issue by phated** [grain-lang/grain#193]
It would be helpful if you could filter/reject a Map in-place. I imagine the API to be something like:

```
let map = Map.fromList([('foo', 1), (bar', 2)]);

Map.filter((key, val) => val == 1, map);
```

I'm unsure if it should use a Boolean or an Option to determine the filters. @ospencer @belph which do you prefer?

**Comment by ospencer** [comment]
I would lean towards `Boolean`, to match the other `filter` operations. `Option` would be a little scary because you could potentially change the value while you were filtering. Could implement that separately as `filterMap` though 👍

**Comment by belph** [comment]
Yeah, I also think that `Boolean` is the way to go. We could have another function in Lists or somewhere else which is more similar to what you describe.

**Comment by phated** [comment]
`filterMap` sounds awesome, however I've been avoiding the term `map` because 1. it's a mutable data structure 2. it is named Map 😜

**Comment by ohana54** [comment]
Attempted to fix in #344
